src/OrchardCore.Modules/OrchardCore.Media/Controllers/AdminController.cs

```diff
@@ -490,5 +506,38 @@ private HashSet<string> GetRequestedExtensions(string exts, bool fallback)
 
             return [];
         }
+
+        private string GetCacheBustingMediaPublicUrl(string path) =>
+            _fileVersionProvider.AddFileVersionToPath(HttpContext.Request.PathBase, _mediaFileStore.MapPathToPublicUrl(path));
+
+        // If a remote storage is used, then we need to preemptively cache the newly uploaded or renamed file. Without
+        // this, the Media Library page will try to load the thumbnail without a cache busting parameter, since
+        // ShellFileVersionProvider won't find it in the local cache.
+        // This is not required for files moved across folders, because the folder will be reopened anyway.
+        private async Task PreCacheRemoteMedia(IFileStoreEntry mediaFile, Stream stream = null)
+        {
+            var mediaFileStoreCache = _serviceProvider.GetService<IMediaFileStoreCache>();
+            if (mediaFileStoreCache == null)
+            {
+                return;
+            }
+
+            Stream localStream = null;
```

Comment:
= stream ?? _mediafile....

Then remove the following if condition.

Reply:
That's not the same. We have two stream variables:

- `stream` coming from the outside; we mustn't dispose that.
- `localStream` if `stream` is `null`; then, we need to use it in place of `stream`, and dispose it.

With such an expression, we wouldn't be able to distinguish between the two, disposing the `Stream` if it comes from the parameter too.